Automattic/woocommerce-payments
# Old WC versions: try loading WCPay if the account is connected

Context: paJDYF-2Dr-p2#comment-8197

#### Changes proposed in this Pull Request

Handle the case new WCPay versions do not support current WC core versions: 

- Continue loading WCPay if the account is connected. Note: at this stage, no main plugin file is loaded, therefore I can not use `WC_Payments_Account::is_stripe_connected`. Instead, I use option_name `wcpay_account_data` (can not use [this constant](https://github.com/Automattic/woocommerce-payments/blob/87c34e476ec97e66f905ab67c49db02fe33c09de/includes/class-wc-payments-account.php#L23-L23) too due to the same loading issue).
- (Extra) add an additional message to suggest merchants using the previous version of WooCommerce Payments.









| Before  | After |
| ------------- | ------------- |
| ![before](https://user-images.githubusercontent.com/10045087/135231002-c82083b7-336a-4f3c-aa68-10c5957a9240.png)  | ![after](https://user-images.githubusercontent.com/10045087/135230988-35002b60-9f4a-4ba4-9b93-5bf1d86fe829.png) |


#### Testing instructions

- Ensure that WCPay account is still connected on your site.
- Edit this line to a version higher in your current Woo version. For example, `5.8`, `6.0`, etc.

https://github.com/Automattic/woocommerce-payments/blob/87c34e476ec97e66f905ab67c49db02fe33c09de/woocommerce-payments.php#L11-L11

- Expect: the `Before` screenshot above, AND no WCPay menu items is loading.
- Apply this PR. 
- Expect: the `After` screenshot above, AND WCPay menu items are still loading.





*

-------------------


**Post merge**
